Origin: upstream https://github.com/apache/trafficserver/pull/5935
Description: Perform a SSL quiet shutdown when close-notify is not sent
--- a/iocore/net/SSLNetVConnection.cc
+++ b/iocore/net/SSLNetVConnection.cc
@@ -907,6 +907,11 @@ SSLNetVConnection::do_io_close(int lerrn
       // Send the close-notify
       int ret = SSL_shutdown(ssl);
       Debug("ssl-shutdown", "SSL_shutdown %s", (ret) ? "success" : "failed");
+    } else {
+      // Request a quiet shutdown to OpenSSL
+        SSL_set_quiet_shutdown(ssl, 1);
+        SSL_set_shutdown(ssl, SSL_RECEIVED_SHUTDOWN | SSL_SENT_SHUTDOWN);
+        Debug("ssl-shutdown", "Enable quiet shutdown");
     }
   }
   // Go on and do the unix socket cleanups
